File: devops-scm-api/src/main/java/com/tencent/devops/scm/api/pojo/Review.java
package com.tencent.devops.scm.api.pojo;

import com.tencent.devops.scm.api.enums.ReviewState;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.NonNull;

import java.util.List;

@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Review {

    @NonNull
    private Long id;
    // 工蜂review事件有iid
    private Integer iid;
    private String body;
    private ReviewState state;
    @NonNull
    private String link;
    private User author;
    private Boolean closed;
    private String title;
    private List<User> reviewers;
    // review相关信息
    private String sourceCommit;
    private String sourceBranch;
    private String sourceProjectId;
    private String targetCommit;
    private String targetBranch;
    private String targetProjectId;
}

File: devops-scm-api/src/main/kotlin/com/tencent/devops/scm/api/pojo/Review.kt
package com.tencent.devops.scm.api.pojo

import com.tencent.devops.scm.api.enums.ReviewState

/**
 * 代码评审信息
 */
data class Review(
    // 评审ID (非空)
    val id: Long,
    // 工蜂评审事件的iid
    val iid: Int,
    // 评审内容
    val body: String? = null,
    // 评审状态
    val state: ReviewState? = null,
    // 评审链接 (非空)
    val link: String,
    // 评审作者
    val author: User? = null,
    // 是否已关闭
    val closed: Boolean? = null,
    // 评审标题
    val title: String,
    // 评审人列表
    val reviewers: List<User>? = null,
    // 源提交
    val sourceCommit: String? = null,
    // 源分支
    val sourceBranch: String? = null,
    // 源项目ID
    val sourceProjectId: String? = null,
    // 目标提交
    val targetCommit: String? = null,
    // 目标分支
    val targetBranch: String? = null,
    // 目标项目ID
    val targetProjectId: String? = null
)
